src/YAKL_defines.h

```diff
@@ -34,6 +34,12 @@
   #define YAKL_CURRENTLY_ON_HOST() (! defined(__SYCL_DEVICE_ONLY__))
   #define YAKL_CURRENTLY_ON_DEVICE() (defined(__SYCL_DEVICE_ONLY__))
 
+  #ifdef __SYCL_DEVICE_ONLY__
```

Comment:
`CONSTANT` doesn't appear to be used anywhere. Is this for future SYCL work?

Reply:
You are right, the `CONSTANT` was for the use case of `sycl::printf`, https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/builtins.hpp#L67. For eg:
```
const CONSTANT char format[] = "KERNEL CHECK FAILED:\n   %s\n   %s\n";
sycl::ext::oneapi::experimental::printf(format,msg);
```
Given that `printf(%s\n")` with `const char *` is not really aligning with the above `sycl::printf` signature